R-packages/evalcast/R/error_measures.R

```diff
@@ -99,6 +99,94 @@ interval_coverage <- function(coverage) {
   }
 }
 
+#' Overprediction component of the weighted interval score
+#' 
+#' Requires symmetric quantile forecasts. Roughly, a penalty for predicted 
+#' quantiles smaller than .5 falling above the observed value.
+#'
+#' @param quantile vector of forecasted quantiles
+#' @param value vector of forecasted values
+#' @param actual_value Actual value.
+#' 
+#' @export
+overprediction <- function(quantile, value, actual_value) {
+  score_func_param_checker(quantile, value, actual_value, "overprediction")
+  if (!is_symmetric(quantile)) {
+    warning(paste0("overprediction/underprediction/sharpness require",
+                   "symmetric quantile forecasts. Using NA."))
+    return(NA)
+  }
+  if (all(is.na(actual_value))) return(NA)
+  actual_value <- unique(actual_value)
+  
+  lower <- value[!is.na(quantile) & quantile < .5]
+  med <- value[find_quantile_match(quantile, 0.5)]
+  
+  if (length(med) > 1L) return(NA)
+  m <- ifelse(length(med == 1L),
+              (med - actual_value) * (med > actual_value),
+              NULL)
+  
+    
+  
+  ans <- mean(c(
+    rep((lower - actual_value) * (lower > actual_value), 2), m))
```

Comment:
Why are we weighting the predicted values double over the median?

Reply:
This is just how WIS gets calculated. Essentially counted once for the lower bound and once for the upper bound. But the median gets only once.